icalevents/icalparser.py

```diff
@@ -282,19 +292,25 @@ def parse_events(content, start=None, end=None, default_span=timedelta(days=7)):
     # assume it applies globally, otherwise UTC
     if len(timezones) == 1:
         cal_tz = gettz(list(timezones)[0])
+        if not cal_tz and str(c['TZID']) in WINDOWS_TO_OLSON:
+            cal_tz = gettz(WINDOWS_TO_OLSON[str(c['TZID'])])
```

Comment:
Tell me if I'm wrong but wouldn't it make sense to move this up to the `if 'X-WR-TIMEZONE' in calendar: ...` clause?

I interpreted the information I found as the following, if `X-WR-TIMEZONE` is used we want to convert it from a standard timezone. Which you do here.

Or must it be used for VTIMEZONE as well?

Reply:
yes - we can move this functionality to a function